src/Microsoft.Framework.FileSystemGlobbing/Internal/PatternContexts/PatternContextRagged.cs

```diff
@@ -81,19 +70,49 @@ public override void PushFrame(DirectoryInfoBase directory)
                 }
             }
 
-            PushFrame(frame);
+            PushDataFrame(frame);
+        }
+
+        public struct FrameData
+        {
+            public bool IsNotApplicable;
+
+            public int SegmentGroupIndex;
+
+            public IList<IPathSegment> SegmentGroup;
+
+            public int BacktrackAvailable;
+
+            public int SegmentIndex;
+        }
+
+        protected IRaggedPattern Pattern { get; }
+
+        protected bool IsStartsWith()
```

Comment:
Also, IsStartsWith, IsEndsWith etc as names do not sound very grammatical. Consider renaming these.